Code/app/src/main/java/ca/ualberta/moodroid/ui/AddMood.java

```diff
@@ -99,111 +91,129 @@ protected void onCreate(Bundle savedInstanceState) {
         scared_button = findViewById(R.id.Scared);
         sick_button = findViewById(R.id.Sick);
 
-        /*
+        /**
          * The purpose of this code is to simply display the mood you tap in the center
          * of the screen, and then take that mood's image, color, and name and bring it to the next screen
          * where you will be able to set the details. Below are a few onclick listeners for
          * each mood button in the wheel. When tapped, they access their own image from the
          * Drawable folder and draw that image in the center.
          */
 
-        annoyed_button.setOnClickListener(new View.OnClickListener() {
+        //get mood models
+        moodService = new MoodService();
+        allMoods = new ArrayList<>();
+        moodService.getAllMoods().addOnSuccessListener(new OnSuccessListener<List<MoodModel>>() {
             @Override
-            public void onClick(View v) {
-                int annoyed_imageRes = getResources().getIdentifier(annoyed_drawable_id, null, getOpPackageName());
-                Drawable res = getResources().getDrawable(annoyed_imageRes);
-                center_button.setImageDrawable(res);
-                center_filled = true;
-                selected_img = annoyed_drawable_id;
-                mood_name = "Annoyed";
-                hex = "#830678";
+            public void onSuccess(List<MoodModel> moodModels) {
+                List<MoodModel> m = moodModels;
+                allMoods.addAll(m);
+                for (MoodModel moodModel : moodModels) {
+                    if (moodModel.getName().equals("Annoyed")) {
```

Comment:
you should use a switch statement here instead

Reply:
changed to switch